### PR TITLE
feat: add nativeWebTapStrict cap

### PR DESCRIPTION
**Pull request by imurchie**
Add a capability, `nativeWebTapStrict`, which skips the attempt to not convert the web coordinates. See https://github.com/appium/appium/issues/14084

First, this makes sure the web coordinate conversion works for newer devices.

### CODE REVIEW

File: lib/commands/web.js [hunk 1]
```diff
@@ -41,17 +44,26 @@ extensions.getSafariIsIphone = _.memoize(async function getSafariIsIphone () {
...
 
-extensions.getSafariIsIphoneX = _.memoize(async function getSafariIsIphone () {
-  try {
-    const script = 'return {height: window.screen.availHeight, width: window.screen.availWidth};';
-    const {height, width} = await this.execute(script);
-    // check for the correct height and width
-    const [portraitHeight, portraitWidth] = height > width ? [height, width] : [width, height];
-    return (portraitHeight === IPHONE_X_HEIGHT && portraitWidth === IPHONE_X_WIDTH) ||
-           (portraitHeight === IPHONE_XR_HEIGHT && portraitWidth === IPHONE_XR_WIDTH);
+extensions.getSafariDeviceSize = _.memoize(async function getSafariDeviceSize () {
```
Comment (mykola-mokhnach) [inline lib/commands/web.js]: what if screen orientation gets changed during the test?

File: lib/commands/web.js [hunk 2]
```diff
@@ -41,17 +44,26 @@ extensions.getSafariIsIphone = _.memoize(async function getSafariIsIphone () {
...
+  return {
+    width: normWidth,
+    height: normHeight,
+  };
+});
 
+extensions.getSafariIsNotched = _.memoize(async function getSafariIsNotched () {
+  try {
+    const {width, height} = await this.getSafariDeviceSize();
+    for (const device of NOTCHED_DEVICE_SIZES) {
```
Comment (mykola-mokhnach) [inline lib/commands/web.js]: `return NOTCHED_DEVICE_SIZES .some(({w, h}) => w === width && h === height)`

File: lib/driver.js [hunk 3]
```diff
@@ -156,12 +155,12 @@ class XCUITestDriver extends BaseDriver {
...
 
   async onSettingsUpdate (key, value) {
-    if (key !== 'nativeWebTap') {
+    if (key !== 'nativeWebTap' && key !== 'nativeWebTapStrict') {
       return await this.proxyCommand('/appium/settings', 'POST', {
         settings: {[key]: value}
       });
     }
-    this.opts.nativeWebTap = !!value;
+    this.opts[key] = !!value;
```
Comment (mykola-mokhnach) [inline lib/driver.js]: I'm not quite sure this is not going to break anything...

**Comment by mykola-mokhnach** [comment]
IMHO, this all logic that we implement on the driver level could be moved to WDA under a custom endpoint. This could speed up the execution a lot, because each proxy call requires additional time (and `click` method is called pretty often). Also, inside WDA we don't have to hardcode many magic constant, but retrieve them directly fro the system instead.

File: lib/driver.js (hunk 3 above)
Comment (imurchie) [reply lib/driver.js]: The only two keys that get through to here are the native web tap ones. This is not a problem.

**Comment by imurchie** [comment]
Feel free to implement it there.

This should all be a last resort anyway. Most of the time atoms are a better way to do things in the web context.

File: lib/desired-caps.js [hunk 4]
```diff
@@ -217,7 +217,10 @@ let desiredCapConstraints = _.defaults({
   },
   appPushTimeout: {
     isNumber: true
-  }
+  },
+  nativeWebTapStrict: {
```
Comment (KazuCocoa) [inline lib/desired-caps.js]: update readme, too?

File: lib/commands/web.js (hunk 1 above)
Comment (imurchie) [reply lib/commands/web.js]: This is just for deciding on the device, and it is normalized to portrait. So it does not matter.

**Pull request merged**